python/pyspark/pandas/sql_formatter.py

```diff
@@ -265,7 +266,10 @@ def _convert_value(self, val: Any, name: str) -> Optional[str]:
             val._to_spark().createOrReplaceTempView(df_name)
             return df_name
         elif isinstance(val, str):
-            return lit(val)._jc.expr().sql()  # for escaped characters.
+            if is_remote():
+                return f"'{val}'"
```

Comment:
I think there is behavior change:

```
In [1]: from pyspark.sql.functions import *

In [2]: val = "a'''c''d"

In [3]: lit(val)._jc.expr().sql()
Out[3]: "'a\\'\\'\\'c\\'\\'d'"
```

Reply:
Added test case with updating the code to handle escaped characters more properly. Thanks!